extractor/src/main/scala/com/digitalasset/extractor/Main.scala

```diff
@@ -18,7 +19,7 @@ object Main extends App with StrictLogging {
 
   logger.trace(s"Parsed config: ${config}")
 
-  val runner = new Extractor(config, target)
+  val runner = new Extractor(config, target, (config, target, ledgerId) => Writer(config, target, ledgerId))
```

Comment:
```suggestion
  val runner = new Extractor(config, target)()
```

And likewise in the other cases.